src/main/kotlin/gg/ingot/iron/transformer/ResultTransformer.kt

```diff
@@ -54,6 +58,39 @@ object ResultTransformer {
         }
     }
 
+    /**
+     * Retrieve the value from the result set for the given field.
+     * Will automatically convert an [Array] into a given [Collection] type if the field is said [Collection].
+     * @param field The field to retrieve the value for.
+     * @return The value from the result set.
+     */
+    private fun ResultSet.retrieveValue(field: EntityField): Any? {
+        val type = field.javaField.type
+
+        if(type.isArray) {
+            return getArray(field.columnName)
+                ?.array
+                ?: return null
```

Comment:
We should opt into throwing instead of returning null since this shouldn't ever fail and if it does, it should be explicit

Reply:
No because if the field is nullable we don't want it to throw, if it's not nullable and for whatever reason we get null back it'll be picked up by the ResultTransformer and throw an error